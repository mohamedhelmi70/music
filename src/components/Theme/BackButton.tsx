import React from 'react';
import {StyleSheet, Pressable} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {isLTR, wp} from '../../constants';
import {useNavigation} from '@react-navigation/native';

function BackButton({color = '#FFF'}) {
    const navigation = useNavigation();

    return (
        <Pressable style={styles.container} onPress={() => navigation.goBack()}>
            <Icon
                name={isLTR ? 'md-arrow-back-sharp' : 'md-arrow-forward-sharp'}
                color={color}
                style={{marginTop: wp('1%')}}
                size={27}
            />
        </Pressable>
    );
}

const styles = StyleSheet.create({
    container: {
        width: wp('14%'),
        alignItems: 'center',
        justifyContent: 'center',
    },
});

export default BackButton;
