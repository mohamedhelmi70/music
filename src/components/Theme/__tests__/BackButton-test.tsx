import React from 'react';
import renderer from 'react-test-renderer';
import BackButton from '../BackButton';

// Describing a test suite
describe('<BackButton />', () => {
    test('Renders Back button correctly', () => {
        const tree = renderer.create(<BackButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
