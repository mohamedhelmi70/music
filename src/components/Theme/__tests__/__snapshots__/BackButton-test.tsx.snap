// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> Renders Back button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "width": 105,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 27,
        },
        Object {
          "marginTop": 7.5,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï„©
  </Text>
</View>
`;
